### ISSUE TITLE
a11y: improve keyboard navigability

### ISSUE THREAD

**Issue by candu** [CityofToronto/bdit_flashcrow#632]
**Description**
During internal testing with the NVDA screen reader, several issues around keyboard navigation, tab ordering, and autofocus were discovered.  This task looks to address those.

**Acceptance Criteria**
- [x] when logged in, user menu should be keyboard navigable;
- [x] location selection:
  - [x] when searching for a location, location search results should be read out as they are focused;
  - [x] when a single location is selected and the user has not entered View Data mode or has collapsed the View Data drawer, the map popup should always show up (so that the "View Data" button can be tabbed to);
  - [x] after clicking "Add Location", focus should go to the "Search for or click..." input box;
- [x] status menu: trying to open the "Assign To" submenu closes the whole menu!
- [x] `FcDataTableRequests`: checkboxes (both "Select All" and individual row) not in tab order!

**Additional Notes**
There are specific problems with the Bulk Request Study wizard and Due Date field that will be covered in separate issues.
